### ISSUE TITLE
Windows build failing

### ISSUE THREAD

**Issue by mindbound** [JuliaIO/ImageMagick.jl#105]
When trying to install ImageMagick.jl on Windows 10, the building process fails with

```
julia> Pkg.build("ImageMagick")
INFO: Building SpecialFunctions
INFO: Building ImageMagick
============================================================================================================[ ERROR: ImageMagick ]============================================================================================================

LoadError: MethodError: no method matching generate_steps(::BinDeps.LibraryDependency, ::BinDeps.CustomPathBinaries, ::Dict{Symbol,Any})
Closest candidates are:
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.AptGet, ::Any) at C:\Users\XXX\.julia\v0.6\BinDeps\src\dependencies.jl:422
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.Yum, ::Any) at C:\Users\XXX\.julia\v0.6\BinDeps\src\dependencies.jl:434
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.Pacman, ::Any) at C:\Users\XXX\.julia\v0.6\BinDeps\src\dependencies.jl:446
  ...
while loading C:\Users\XXX\.julia\v0.6\ImageMagick\deps\build.jl, in expression starting on line 118

==============================================================================================================================================================================================================================================ERROR: UndefVarError: BinDeps not defined
Stacktrace:
 [1] deserialize_module(::SerializationState{IOStream}) at .\serialize.jl:754
 [2] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:695
 [3] deserialize(::SerializationState{IOStream}) at .\serialize.jl:634
 [4] deserialize_datatype(::SerializationState{IOStream}, ::Bool) at .\serialize.jl:968
 [5] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:674
 [6] deserialize(::SerializationState{IOStream}) at .\serialize.jl:634
 [7] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:681
 [8] deserialize(::SerializationState{IOStream}, ::DataType) at .\serialize.jl:1075
 [9] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:687
 [10] deserialize(::SerializationState{IOStream}, ::DataType) at .\serialize.jl:1075
 [11] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:687
 [12] (::Base.Pkg.Entry.##54#55{Dict{Any,Any}})(::IOStream) at .\pkg\entry.jl:639
 [13] open(::Base.Pkg.Entry.##54#55{Dict{Any,Any}}, ::String, ::String) at .\iostream.jl:152
 [14] build!(::Array{String,1}, ::Dict{Any,Any}, ::Set{Any}) at .\pkg\entry.jl:636
 [15] build(::Array{String,1}) at .\pkg\entry.jl:650
 [16] (::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}})() at .\pkg\dir.jl:36
 [17] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}}, ::String) at .\file.jl:59
 [18] #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at .\pkg\dir.jl:36
 [19] build(::String, ::Vararg{String,N} where N) at .\pkg\pkg.jl:254
```

I'm using
```
julia> versioninfo()
Julia Version 0.6.0
Commit 903644385b* (2017-06-19 13:05 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-3610QM CPU @ 2.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, ivybridge)
```

**Comment by musm** [comment]
Looks related to SpecialFunctions not ImageMagick

**Comment by mobius-eng** [comment]
I get the same error from building ImageMagick:

```
julia> Pkg.build("ImageMagick")
INFO: Building WinRPM
INFO: Downloading https://cache.julialang.org/http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_Leap_42.2/repodata/repomd.xml
INFO: Downloading https://cache.julialang.org/http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_Leap_42.2/repodata/repomd.xml
INFO: Building FFTW
INFO: Building SpecialFunctions
INFO: Building ImageMagick
=================================================[ ERROR: ImageMagick ]=================================================

LoadError: MethodError: no method matching generate_steps(::BinDeps.LibraryDependency, ::BinDeps.CustomPathBinaries, ::Dict{Symbol,Any})
Closest candidates are:
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.AptGet, ::Any) at C:\Users\alexe\.julia\v0.6\BinDeps\src\dependencies.jl:446
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.Yum, ::Any) at C:\Users\alexe\.julia\v0.6\BinDeps\src\dependencies.jl:458
  generate_steps(::BinDeps.LibraryDependency, ::BinDeps.Pacman, ::Any) at C:\Users\alexe\.julia\v0.6\BinDeps\src\dependencies.jl:470
  ...
while loading C:\Users\alexe\.julia\v0.6\ImageMagick\deps\build.jl, in expression starting on line 122

========================================================================================================================
ERROR: UndefVarError: BinDeps not defined
Stacktrace:
 [1] deserialize_module(::SerializationState{IOStream}) at .\serialize.jl:757
 [2] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:698
 [3] deserialize(::SerializationState{IOStream}) at .\serialize.jl:637
 [4] deserialize_datatype(::SerializationState{IOStream}, ::Bool) at .\serialize.jl:972
 [5] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:677
 [6] deserialize(::SerializationState{IOStream}) at .\serialize.jl:637
 [7] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:684
 [8] deserialize(::SerializationState{IOStream}, ::DataType) at .\serialize.jl:1079
 [9] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:690
 [10] deserialize(::SerializationState{IOStream}, ::DataType) at .\serialize.jl:1079
 [11] handle_deserialize(::SerializationState{IOStream}, ::Int32) at .\serialize.jl:690
 [12] (::Base.Pkg.Entry.##54#55{Dict{Any,Any}})(::IOStream) at .\pkg\entry.jl:639
 [13] open(::Base.Pkg.Entry.##54#55{Dict{Any,Any}}, ::String, ::String) at .\iostream.jl:152
 [14] build!(::Array{String,1}, ::Dict{Any,Any}, ::Set{Any}) at .\pkg\entry.jl:636
 [15] build(::Array{String,1}) at .\pkg\entry.jl:650
 [16] (::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}})() at .\pkg\dir.jl:36
 [17] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}}, ::String) at .\file.jl:59
 [18] #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at .\pkg\dir.jl:36
 [19] build(::String, ::Vararg{String,N} where N) at .\pkg\pkg.jl:254

julia> Pkg.build("SpecialFunctions")
INFO: Building SpecialFunctions

julia>
```

SpecialFunctions build on its own is fine.

```
Julia Version 0.6.2
Commit d386e40c17* (2017-12-13 18:08 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, broadwell)
```

**Comment by mobius-eng** [comment]
I found the line on which it breaks in `build.jl`:

```julia
@BinDeps.install Dict([(:libwand, :libwand)])
```

**Comment by SimonDanisch** [comment]
Should be fixed by #121! Please try again on master!
